doc/thingpedia-tutorial-linkedin.md

```diff
@@ -0,0 +1,176 @@
+about the device:
+
+- ID: `<your-name>.linkedin` (Each device in Thingpedia needs an unique ID, so use your name or email address 
+in the device name to make sure it won't conflict with others)
+- Name: `My LinkedIn`
+- Description: `LinkedIn Account in Almond`
+- Category: `Social Network`
+- Icon: choose a PNG file you like (512x512 resolution is recommended)
+- JS code: upload a file named `index.js` with the following code.
+```javascript
+"use strict";
+
+const Tp = require('thingpedia');
+
+const PROFILE_URL = 'https://api.linkedin.com/v1/people/~:(id,formatted-name,headline,industry,specialties,positions,picture-url)?format=json';
+const SHARE_URL = 'https://api.linkedin.com/v1/people/~/shares?format=json';
+
+module.exports = class LinkedinDevice extends Tp.BaseDevice {
+    /*
+    runOAuth2 specifies how the authentication works in detail 
+    */
+    static get runOAuth2() {
+        return Tp.Helpers.OAuth2({
+            authorize: 'https://www.linkedin.com/uas/oauth2/authorization',
+            get_access_token: 'https://www.linkedin.com/uas/oauth2/accessToken',
+            set_state: true,
+
+            callback(engine, accessToken, refreshToken) {
+                const auth = 'Bearer ' + accessToken;
+                return Tp.Helpers.Http.get('https://api.linkedin.com/v1/people/~:(id,formatted-name)?format=json',
+                                           { auth: auth,
+                                             accept: 'application/json' }).then((response) => {
+                    const parsed = JSON.parse(response);
+                    return engine.devices.loadOneDevice({ kind: 'com.linkedin',
+                                                          accessToken: accessToken,
+                                                          refreshToken: refreshToken,
+                                                          userId: parsed.id,
+                                                          userName: parsed.formattedName
+                                                        }, true);
+                });
+            }
+        });
+    }
+
+    /* 
+    A user might have multiple accounts for non-public services.
+    Thus we need to have a unique ID to identify different instance of the class.
+    And optionally, we give each instance a different name and description, so
+    that users can easily tell which account a device instance associates with 
+    in their device list. 
+    */
+    constructor(engine, state) {
+        super(engine, state);
+
+        this.uniqueId = 'com.linkedin-' + this.state.userId;
+        this.name = "LinkedIn Account of %s".format(this.state.userName);
+        this.description = "This is your LinkedIn account";
+    }
+
+    /*
+    A query function called "get_profile"
+    The "get_" prefix indicates this is a query, not an action
+    */
+    get_get_profile() {
+        return Tp.Helpers.Http.get(PROFILE_URL, {
```

Comment:
More comments? Say that this function calls an HTTP request, comment on the options we pass to the HTTP helpers, and maybe add a reference to the HTTP helpers page